config/defaults.mk

```diff
@@ -95,6 +95,10 @@ else
    # Silence unused command line argument warnings when generating dependencies
    # with mpicxx and clang
    DEP_FLAGS := -Wno-unused-command-line-argument $(DEP_FLAGS)
+   # Silence "ignoring duplicate libraries" warnings on new (Xcode 15) linker
+   ifneq (,$(findstring PROJECT:dyld,$(shell ld -v 2>&1)))
+      LDFLAGS = -Xlinker -no_warn_duplicate_libraries
```

Comment:
It's probably better to use a different variable here: e.g. `LDFLAGS_INTERNAL` or `LDFLAGS_PRIVATE` because `LDFLAGS` is a commonly used variable that users can set. The flags set here should be added to the link flags similar to `LDFLAGS`: https://github.com/mfem/mfem/blob/37e03ab550e83cda2214e3b230e2aea5b24328c9/makefile#L367 and it will be good to update the comment here too: https://github.com/mfem/mfem/blob/37e03ab550e83cda2214e3b230e2aea5b24328c9/makefile#L196-L197

Reply:
Done in 22ec7e7ad